app/src/main/java/org/oppia/app/player/audio/CellularDataDialogFragment.kt

```diff
@@ -0,0 +1,55 @@
+package org.oppia.app.player.audio
+
+import android.app.Dialog
+import android.content.Context
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import android.widget.CheckBox
+import org.oppia.app.R
+
+/**
+ * DialogFragment that controls user preference for dialog box
+ */
+class CellularDataDialogFragment : DialogFragment() {
+  private var doNotShowAlert: Boolean = false
+
+  companion object {
+    lateinit var cellularDataInterface: CellularDataInterface
+    /**
+     * This function is responsible for displaying content in DialogFragment
+     * @param cellularDataInterface: [CellularDataInterface] to send data back to parent
+     * @return [CellularDataDialogFragment]: DialogFragment
+     */
+    fun newInstance(
+      cellularDataInterface: CellularDataInterface
+    ): CellularDataDialogFragment {
+      this.cellularDataInterface = cellularDataInterface
+      return CellularDataDialogFragment()
+    }
+  }
+
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    val view = activity!!.layoutInflater.inflate(R.layout.cellular_data_dialog, null)
+    val checkBox = view.findViewById<CheckBox>(R.id.cellular_data_dialog_checkbox)
+
+    checkBox.setOnCheckedChangeListener { buttonView, isChecked ->
+      doNotShowAlert = isChecked
```

Comment:
Do we need to retain this state on configuration changes?

Reply:
Actually, even if we save this or not this is blocked on configuration change support for `showCellularDataDialogFragment()`.

After running the app and changing the orientation of app, the app crashes with following message:
`FragmentManager is already executing transactions`
on `fragmentTransaction` in `showCellularDataDialogFragment`, so before fixing this `doNotShowAlert` variable, I will have to first fix this dialog.

Reply:
As discussed via chat, this is likely because you're creating the fragment in onCreateView(), and that will be called again during config changes. We can't perform fragment transactions (e.g. via showNow()) within another transaction (e.g. onCreateView() during config change).

Moving management of showing the dialog to a button in StateFragment should address this issue since it won't be attempting to show the dialog in a lifecycle method.

Reply:
Thanks for summarising this here.

Reply:
This approach is correct and is working correctly as per above discussed implementation.